presto-main/src/main/java/com/facebook/presto/type/DecimalCasts.java

```diff
@@ -467,6 +467,16 @@ public static Slice realToLongDecimal(long value, long precision, long scale, Bi
         return encodeUnscaledValue(decimalBigInteger);
     }
 
+    private static BigDecimal doubleToBigDecimal(double value, long precision, long scale)
+    {
+        try {
+            return new BigDecimal(value);
+        }
+        catch (NumberFormatException ex) { // inf or nan
```

Comment:
I wanted to avoid duplication because `BigDecimal` does that check internally, so it's BigDecimal's c-tor responsibility to check boundaries.